### PR TITLE
Run `make upgrade`

### PR DESCRIPTION
**Pull request by feanil**
This is prep work so I can unpin DRF in here so we can upgrade it in edx-platform.

### CODE REVIEW

File: requirements/doc.in [hunk 1]
```diff
@@ -3,5 +3,5 @@
 -r test.txt               # Core and test dependencies so sphinx-apidoc can import the code to document
 
 readme_renderer           # Validates README.rst for usage on PyPI
-Sphinx                    # Documentation builder
+Sphinx<2.0.0              # Documentation builder, 2.0.0 drops support for python 2.7
```
Comment (jmbowman) [inline requirements/doc.in]: Is this a good time to introduce a `constraints.txt` file?  Example: https://github.com/edx/edx-platform/commit/75b051b336aa308f7b8735ab0ddbacaf0563654f#diff-294960731653f2fdb7b827504766ff5d

File: requirements/test.in [hunk 2]
```diff
@@ -7,3 +7,6 @@ pytest-django             # pytest extension for better Django support
 ddt                       # Run a test case multiple times with different input
 freezegun                 # Allows tests to mock the output of assorted datetime module functions
 mock
+
+more-itertools<6.0.0      # A dependency of pytest.  Pytest doesn't constrain it and 6.0.0 onward
+                          # only works with python 3
```
Comment (jmbowman) [inline requirements/test.in]: This would also go in `constraints.txt`.

File: requirements/doc.in (hunk 1 above)
Comment (feanil) [reply requirements/doc.in]: @jmbowman I considered that, but it looked like in platform we had just a single shared constraints file, whereas here, I would want to make one for docs and one for test?  That seemed heavier weight than necessary but there is something to be said for consistency.  Have you thought about how you'd like to do this when you need to limit only specific `.in` files?x

Comment (feanil) [reply requirements/doc.in]: Done.

File: requirements/test.in (hunk 2 above)
Comment (feanil) [reply requirements/test.in]: Done.

**Comment by jmbowman** [comment]
I've started using GitHub's new-ish "Squash and Merge" (from the dropdown by the merge button) for cases like this; preserves the commit history in the PR but keeps the master branch clean.

**Pull request merged**